src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -158,6 +158,14 @@ internal virtual ImmutableArray<LocalSymbol> GetDeclaredLocalsForScope(CSharpSyn
         }
 
         /// <summary>
+        /// Get local functions declared immediately in scope represented by the node.
+        /// </summary>
+        internal virtual ImmutableArray<LocalFunctionSymbol> GetDeclaredLocalFunctionsForScope(CSharpSyntaxNode node)
```

Comment:
What is the purpose of providing the `node` parameter here?  In all cases I saw below either 

a. The node is exactly what we expect and we return a value not derived from `node` 
b. The node is not what we expect and we throw

I don't quite see the value of the `node` with this logic.